src/drivers/rotor_frequency/pcf8583/pcf8583.cpp

```diff
@@ -0,0 +1,702 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2013-2016 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/**
+ * @file pcf8583.cpp
+ *
+ * @author Vít Hanousek <slimonslimon@gmail.com>
+ *
+ * Driver for Main Rotor speed sensor using MLAB.CZ RTC003A I2C counter.
+ */
+
+#include <px4_platform_common/px4_config.h>
+#include <px4_platform_common/defines.h>
+#include <px4_platform_common/getopt.h>
+#include <px4_platform_common/workqueue.h>
+
+#include <lib/parameters/param.h>
+
+#include <board_config.h>
+
+#include <px4_log.h>
+
+#include <drivers/device/i2c.h>
+
+
+#include <sys/types.h>
+/*#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <semaphore.h>
+
+#include <fcntl.h>
+#include <poll.h>
+#include <stdio.h>
+#include <math.h>
+#include <unistd.h>
+#include <vector>
+
+
+
+#include <perf/perf_counter.h>
+*/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include <uORB/uORB.h>
+#include <uORB/topics/rotor_frequency.h>
+
+#include <drivers/drv_hrt.h>
+#include <drivers/drv_sensor.h>
+
+
+/* Configuration Constants */
+#define PCF8583_BUS_DEFAULT 		      PX4_I2C_BUS_EXPANSION
+#define PCF8583_DEVICE_PATH	         "/dev/pcf8583"
```

Comment:
The dev and ioctls can be gutted. Everything is done over uORB and parameters.

Reply:
Done in last commit. But these defines remains for I2C class..